### PR TITLE
[Auth 4] refactor RefreshTokenAccount 

### PR DESCRIPTION
**Pull request by andxu**
1. refactor initializeCredentials to createTokenCredential
2. remove RefreshTokenAccount and use RefreshTokenTenantCredential instead

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java [hunk 1]
```diff
@@ -51,19 +57,13 @@ private static AccessToken fromAuthenticationResult(AuthenticationResult authent
...
-                                                  String tenantId,
-                                                  String refreshToken,
-                                                  String resource) throws LoginFailureException, MalformedURLException {
+    private AuthenticationResult authenticate(String tenantId, String resource) throws LoginFailureException, MalformedURLException {
         AuthenticationContext context;
         AuthenticationResult result;
-        String authorityUrl = env.getActiveDirectoryEndpoint().replaceAll("/+$", "") + "/" + tenantId;
-        ExecutorService service = null;
+        String authorityUrl = authority + "/" + tenantId;
+        ExecutorService service = Executors.newFixedThreadPool(1);
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:64]: handle `ExecutorService` lifecycle

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java [hunk 2]
```diff
@@ -51,19 +57,13 @@ private static AccessToken fromAuthenticationResult(AuthenticationResult authent
         return new AccessToken(authenticationResult.getAccessToken(), expiresOnDate);
     }
 
-    private static AuthenticationResult authorize(AzureEnvironment env,
-                                                  String clientId,
-                                                  String tenantId,
-                                                  String refreshToken,
-                                                  String resource) throws LoginFailureException, MalformedURLException {
+    private AuthenticationResult authenticate(String tenantId, String resource) throws LoginFailureException, MalformedURLException {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:60]: convert to `AccessToken`

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java [hunk 3]
```diff
@@ -7,36 +7,42 @@
...
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
+import java.util.Map;
+import java.util.concurrent.*;
 
-public class RefreshTokenCredentialFactory {
-    public static TokenCredential fromRefreshToken(AzureEnvironment env, String clientId, String tenantId, String refreshToken) {
-        return buildRefreshTokenCredential(env, clientId, tenantId, refreshToken);
+@AllArgsConstructor
+public class RefreshTokenTenantCredential extends TenantCredential {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:27]: migrate TenantCredential to `CredentialManager`
and add `listTenants` method into `CredentialManager`

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java (hunk 2 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:60]: will do the inner interface update in later PR

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java (hunk 3 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:27]: will refactor later

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java (hunk 1 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/refresktoken/RefreshTokenTenantCredential.java:64]: in  finally

**Review by Flanker32 — DISMISSED** [review]
Please address the comment in later PR

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**